server/src/main/java/org/elasticsearch/search/lookup/SearchLookup.java

```diff
@@ -53,7 +53,22 @@ public SearchLookup(
     ) {
         this.fieldTypeLookup = fieldTypeLookup;
         this.fieldChain = Collections.emptySet();
-        this.sourceLookup = new SourceLookup();
+        this.sourceLookup = new SourceLookup.Static();
+        this.fieldDataLookup = fieldDataLookup;
+    }
+
+    public SearchLookup(
+        Function<String, MappedFieldType> fieldTypeLookup,
+        BiFunction<MappedFieldType, Supplier<SearchLookup>, IndexFieldData<?>> fieldDataLookup,
+        boolean canReloadSource
+    ) {
```

Comment:
I tend not to like adding these extra ctors if I can get away without it. Is this temporary? I think we may have talked about this but I'm forgetting.

Either way, probably worth copying the javadoc from above.

Reply:
From a quick look it appears that all of the uses of the extra constructor are in test code, which makes me agree with Nik.  Let's nuke it - it makes the PR a bit noisier but I think it will save confusion further down the line.